### DISCUSSION TITLE
C/C++语言QUIC接口中发送报文on_packets_send参数count是如何控制的？

### DISCUSSION THREAD

**Discussion by LengM** [Tencent/tquic#380] [Q&A (问题和回答)]
C/C++语言QUIC接口中发送报文on_packets_send参数count是如何控制的？
因为想要进行批量提交在此回调中的使用sendmmsg。
![image](https://github.com/user-attachments/assets/4311565a-7829-4bfc-ae66-4ff8e2359d6f)

**Comment by iyangsj** [comment]
@LengM 批量待发送的报文，通过`pkts` 和 `count`两个参数来指定。 `pkts`代表动态数组的起始地址，其元素类型是 struct `quic_packet_out_spec_t`，元素个数是`count`

**Reply by LengM** [reply]
on_packets_send不是回调函数吗？我在代码中并没有直接调用该方法，所以无法传参，还是说我可以自己调用？

**Reply by iyangsj** [reply]
`on_packets_send`是由应用程序提供给TQUIC库的回调函数，TQUIC会按需调用`on_packets_send`来尝试发送，TQUIC会通过`pkts`和 `count`参数来指定一批待发送报文及数量。

应用程序需要实现该回调函数，执行真正的发送。


简单的示例参见：https://github.com/tquic-group/tquic-example-c/blob/main/simple_client.c#L91
注意：这个简单示例中只使用了sendmsg来逐一发送报文到网络中，可按需改造为使用sendmmsg来批量发送
